### DISCUSSION TITLE
Why Vite does not insert polyfill code when using Babel plugin that does not use configFile

### DISCUSSION THREAD

**Discussion by gofeel8** [vitejs/vite-plugin-react#198] [Q&A]
When I set the configuration related to the babel preset in the config file(babel.config.json) and build it, the polyfill code is inserted well

```
//vite.config.js
export default defineConfig({
  plugins: [
    react({ babel: { configFile: true } }),
  ],
});
```
```
//babel.config.json
{
  "presets": [
    [
      "@babel/preset-env",
      {
        "targets": {
          "safari": "7"
        },
        "modules": false,
        "useBuiltIns": "usage",
        "corejs": "3.31"
      }
    ]
  ]
}

```
However, when I write directly a configuration related to the babel and build it without using the configFile, it doesn't insert the polyfill code
```
//vite.config.js
export default defineConfig({
  plugins: [
    react({
      babel: {
        presets: [
          [
            "@babel/preset-env",
            {
              targets: {
                safari: "7",
              },
              modules: false,
              useBuiltIns: "usage",
              corejs: "3.31",
            },
          ],
        ],
      },
    }),
  ],
});

```
However, polyfill is applied to dev-server well

Why can't the polyfill code be inserted when I build? 

Here is Sample code
https://stackblitz.com/edit/vitejs-vite-fth3nu?file=babel.config.json,vite.config.js
Try testing by changing the comment & yarn build

and 
Comparison of build results
https://www.diffchecker.com/QqcVUFhQ/

**Comment by ArnaudBarre** [comment]
Thanks for the repro. This was fixed in https://github.com/vitejs/vite-plugin-react/commit/6e2e0a406e8478002dc0ab4a5104d370d4afc7b3. I will do a release tomorrow

**Comment by ArnaudBarre** [comment]
Released as v4.0.4!
